enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,283 @@
+3. The cluster administrator can get the merged cluster scope policies from `status` of ImagePolicy instances. 
+4. The cluster administrator can delete the signature verification configuration by deleting its ImagePolicy instances. 
+
+**application administrator** is a human user responsible for
+adding signature verification configurations for specific namespace.
+
+1. The application administrator creates ImagePolicy yaml file for namespace that is not `namespace: opesnhift-config`
+2. The application administrator creates ImagePolicy instance using `oc create -f imagepolicy.yaml`.
+3. The application administrator can get the merged cluster policy and policy for certain namespace from `status` of ImagePolicy instances. 
+4. The application administrator can delete the signature verification configuration by deleting its ImagePolicy instances.
+5. The application administrator cannot override cluster scoped policies because they will treated with a higher priority.
+
+### API Extensions
+
+#### Type definitions
+
+```go
+type ImagePolicySpec struct {
+	// image defines the list of images/repositories assigned to a policy.
+	Images []ImageScope `json:"images"`
+	// policy defines the verification policy
+	Policy Policy `json:"policy"`
+}
+
+// ImageScope is the item of the images list.
+// https://go.dev/play/p/FfL_3bpqSX5
+// +kubebuilder:validation:Pattern=`^\*(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+$|^((?:(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9])(?:(?:\.(?:[a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]))+)?(?::[0-9]+)?/)?[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?(?:(?:/[a-z0-9]+(?:(?:(?:[._]|__|[-]*)[a-z0-9]+)+)?)+)?)(?::([\w][\w.-]{0,127}))?(?:@([A-Za-z][A-Za-z0-9]*(?:[-_+.][A-Za-z][A-Za-z0-9]*)*[:][[:xdigit:]]{32,}))?$`
+type ImageScope string
+
+type Policy struct {
+	// OIDCIssuer contains the expected OIDC issuer.
+	// Requires SubjectEmail to be set and KeyData to be empty.
+	// Example: "https://expected.OIDC.issuer/"
+	OIDCIssuer string `json:"oidcIssuer,omitempty"`
+
+	// SubjectEmail holds the email address of the subject.
+	// Example: "expected-signing-user@example.com"
+	// Requires OIDCIssuer to be set and KeyData to be empty.
+	SubjectEmail string `json:"subjectEmail,omitempty"`
+
+	// KeyData contains inline base64 data of the public key.
+	// Can be empty if the image got signed keyless.
+	// Requires OIDCIssuer and SubjectEmail to be empty.
+	KeyData string `json:"keyData,omitempty"`
+
+	// SignedIdentity specifies what image identity the signature
+	// claims about the image.
+	SignedIdentity Identity `json:"signedIdentity,omitempty"`
+}
+
+type Identity struct {
+	// Defaults to "matchRepository"
+	IdentityMatchPolicy IdentityMatchPolicy `json:"identityMatchPolicy,…"`
+	DockerRepository    string              `json:"dockerRepository,omitempty"`
+}
+
+type IdentityMatchPolicy string
+
+const (
+	IdentityMatchPolicyMatchRepository = "matchRepository"
```

Comment:
I don’t care too much about exposing all the `signedIdentity` options; but the default of `matchRepoDigestOrExact` should also be the default for OpenShift, and `remapIdentity` is much more convenient than `exactRepository`, so I think `remapIdentity` should either be supported out of the gate, or it should at least be clear how that is going to be added.

Reply:
hmm, the containers/image mentioned the cosign signature only contains the repository, does this mean the default should be `matchRepository` since only cosign is supported?

> cosign-created signatures only contain a repository, so only matchRepository and exactRepository can be used

Reply:
I don’t think so; I’d much prefer for `cosign` to change. I.e. default to matching that includes tags, but allow OpenShift users to define a cosign-compatible policy with `matchRepository` / `exactRepository`.

Compare also https://github.com/containers/image/issues/1952 , as a start of the upstream effort to treat signed identities as a critical signature feature.

Reply:
Updated the comment to point out the default `matchRepoDigestOrExact` and support `remapIdentity`.